### PR TITLE
Add dapr user agent for Azure Components

### PR DESCRIPTION
**Pull request by berndverst**
# Description

Set custom user agent for `dapr` to be used with various Azure services.

This change prepends `dapr-$version` to the user Agent, or sets this as the user Agent where no other value exists.

## Issue reference

#1014 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly

The upstream components already have tests for setting custom user agents. No additional tests needed here.

### CODE REVIEW

**Comment by berndverst** [comment]
FYI @greenie-msft

**Review by CodeMonkeyLeet — COMMENTED** [review]
Code changes LGTM, not clear if it addresses #1014 as intended though? @greenie-msft, what's the expectation here?

**Comment by berndverst** [comment]
@CodeMonkeyLeet it does address that issue for Service Bus and Event Hubs, but not other services.

We have an email thread with that service team where we discussed this in depth:
 
TL;DR -- Service Bus and Event Hubs teams receive a product, version, user agent and a few other identifiers. An example they provided was the Python SDK.
 
@greenie-msft and I looked at the Python SDK in depth to see just how this is sent. It is transmitted via the AMQP message metadata properties -- this itself is set globally by the sender client constructor -- so the implementation is very specific to Service Bus and also Event Hubs.
 
The GO SDK for these services (just like the Python one) has the ability to prepend to the user agent that is transmitted. This change allows us to add a dapr specific identifier.

Now the much harder problem -- completely out of scope of this PR or anything on our (Dapr) team -- getting the logs from the Microsoft internal Service Bus / Event Hubs distributing tracing log store (Geneva). The team said they likely need to implement something to directly filter and forward events on their end. Something they need for business reasons anyway and this is on their backlog.

File: pubsub/azure/eventhubs/eventhubs.go [hunk 1]
```diff
@@ -157,7 +158,8 @@ func (aeh *AzureEventHubs) Init(metadata pubsub.Metadata) error {
 		return err
 	}
 	aeh.metadata = m
-	hub, err := eventhub.NewHubFromConnectionString(aeh.metadata.connectionString)
+	hub, err := eventhub.NewHubFromConnectionString(aeh.metadata.connectionString,
```
Comment (artursouza) [inline pubsub/azure/eventhubs/eventhubs.go:161]: Do we want to include version too? Maybe follow the `<product>/<version>` pattern, similar as in https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

**Review by artursouza — CHANGES_REQUESTED** [review]

File: pubsub/azure/eventhubs/eventhubs.go (hunk 1 above)
Comment (berndverst) [reply pubsub/azure/eventhubs/eventhubs.go:161]: Yes we do -- but how can I get the version? :D This is in components contrib after all.

Comment (berndverst) [reply pubsub/azure/eventhubs/eventhubs.go:161]: @artursouza also this isn't a HTTP header, so other user agents don't apply. The way these SDKs are implemented what is sent will be `$ourstring/$theirstring`

Comment (artursouza) [reply pubsub/azure/eventhubs/eventhubs.go:161]: We don't need to match the HTTP header but it is just as inspiration. The version is important. We already set the version in the dapr build, we can add the version to another public property too, this time in components contrib. Take a look at the Makefile in dapr/dapr repo.

**Review by artursouza — APPROVED** [review]
